tests/FakeItEasy.Specs/AssignsOutAndRefParametersSpecs.cs

```diff
@@ -2,122 +2,175 @@ namespace FakeItEasy.Specs
 {
     using System;
     using System.Diagnostics.CodeAnalysis;
+    using FakeItEasy.Configuration;
     using FakeItEasy.Tests.TestHelpers;
     using FluentAssertions;
     using Xbehave;
 
-    public interface IHaveAnOut
+    [SuppressMessage("Microsoft.Naming", "CA1702:CompoundWordsShouldBeCasedCorrectly", MessageId = "WithOut", Justification = "That's two words, not one")]
+    [SuppressMessage("Microsoft.Design", "CA1045:DoNotPassTypesByReference", MessageId = "1#", Justification = "Required for testing.")]
+    [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "2#", Justification = "Required for testing.")]
+    public delegate void VoidDelegateWithOutAndRefParameters(int byValueParameter, ref int byRefParameter, out int outParameter);
+
+    [SuppressMessage("Microsoft.Naming", "CA1702:CompoundWordsShouldBeCasedCorrectly", MessageId = "WithOut", Justification = "That's two words, not one")]
+    [SuppressMessage("Microsoft.Design", "CA1045:DoNotPassTypesByReference", MessageId = "1#", Justification = "Required for testing.")]
+    [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "2#", Justification = "Required for testing.")]
+    public delegate int NonVoidDelegateWithOutAndRefParameters(int byValueParameter, ref int byRefParameter, out int outParameter);
+
+    public interface IHaveARef
     {
-        [SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "0#", Justification = "Required for testing.")]
-        void MightReturnAKnownValue(out string andThisIsWhoReallyDidIt);
+        [SuppressMessage("Microsoft.Design", "CA1045:DoNotPassTypesByReference", MessageId = "0#", Justification = "Required for testing.")]
+        void MightReturnAKnownValue(ref string andThisIsWhoReallyDidIt);
     }
 
     public static class AssignsOutAndRefParametersSpecs
     {
+        const string Condition = "someone_else";

```

Comment:
Actually, I don't think the condition really matters. The out parameter *is* set to the `KnownOutput`, which proves that the lambda was executed; I think it's all that matters. And yes, the tests could be made simpler by removing the condition (and then I wouldn't have needed to change `out` to `ref`...)